clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -303,7 +303,8 @@ def TypeInfoAttr : CIR_Attr<"TypeInfo", "typeinfo", [TypedAttrInterface]> {
 
     The verifier enforces that the output type is always a `!cir.struct`,
     and that the ArrayAttr element types match the equivalent member type
-    for the resulting struct.
+    for the resulting struct, i.e, a GlobalViewAttr for symbol reference or
+    an IntAttr for flags.
```

Comment:
I still would prefer not to use an IntAttr but a enum kind attr that contains BCTI_Virtual, BCTI_Public, etc (and incrementally add the enums as we get usecases). Using `IntAttr` feels too opaque and already LLVM-like. Are there restrictions to doing such approach?